### PR TITLE
Add grammar files for alter index

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Add grammar files for Alter Index. Implementation will be in a later PR.
This PR is to allow new grammar files to sync to sql plugin.

### Issues Resolved
* #278 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4 [hunk 1]
```diff
@@ -118,6 +133,11 @@ describeMaterializedViewStatement
     : (DESC | DESCRIBE) MATERIALIZED VIEW mvName=multipartIdentifier
     ;
 
+alterMaterializedViewStatement
+    : ALTER MATERIALIZED VIEW mvName=multipartIdentifier
+        WITH LEFT_PAREN propertyList RIGHT_PAREN)
```
[the remark this answers is not in the document]
Comment (seankao-az) [reply flint-spark-integration/src/main/antlr4/FlintSparkSqlExtensions.g4:138]: true. nice catch

**Review by dai-chen — APPROVED** [review]

**Comment by seankao-az** [comment]
Thanks for the review. Could you help merge it, thanks! @dai-chen

**Review by noCharger — COMMENTED** [review]
Just confirming, `With` clause is required instead of optional?

**Comment by seankao-az** [comment]
> Just confirming, `With` clause is required instead of optional?

Yes. Alter Index is used for changing the options in With clause.

**Review by noCharger — APPROVED** [review]

**Pull request merged**